### ISSUE TITLE
[rush] watchForChanges doesn't work for Windows

### ISSUE THREAD

**Issue by bagermen** [microsoft/rushstack#4031]
## Summary

watchForChanges doesn't work for Windows and, I guess, for iOS as well

## Repro steps
1. Open repository using Windows
2. add build command (bulk or phased) with `watchOption` 
3. execute `rush build --watch`
4. try to change any file in the project

**Expected result:** changes are fetched by rush
 **Actual result:** Nothing happens

## Details

The issue is inside class ProjectWatcher of @microsoft/rush-lib package.
Git may use `git-fsmonitor` which changes files all the time inside` .git` folder. This breaks this part of code

https://github.com/microsoft/rushstack/blob/da214d030c0bec2c6bbd47e82acd92272e1f25c3/libraries/rush-lib/src/logic/ProjectWatcher.ts#L127-L132

In other words, `timeout `may exists and not a safe way for making decisions

## Standard questions

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.95.0 |
| `rushVersion` from rush.json? | 5.95.0 |
| `useWorkspaces` from rush.json? | Yes |
| Operating system? | Windows  |
| Would you consider contributing a PR? | Not now |
| Node.js version (`node -v`)? | 18.15.0 |

**Comment by octogonz** [comment]
@dmichon-msft is this a recent regression?

**Comment by octogonz** [comment]
Do we know which version of Rush introduced the regression?
